scripts/spec_xml/generate_spec_xml.py

```diff
@@ -94,5 +102,27 @@ def scrape_all_clusters(dir: str, exclude_list: list[str] = []) -> None:
         output.write(sha)
 
 
+def scrape_device_types(scraper, spec_root, output_dir, dry_run):
+    device_type_dir = os.path.abspath(os.path.join(spec_root, 'src', 'device_types'))
+    device_types_output_dir = os.path.abspath(os.path.join(output_dir, 'device_types'))
+    clusters_output_dir = os.path.abspath(os.path.join(output_dir, 'clusters'))
+
+    if not os.path.exists(device_types_output_dir):
+        os.makedirs(device_types_output_dir)
+
+    def scrape_device_type(filename: str) -> None:
+        xml_path = get_xml_path(filename, device_types_output_dir)
+        print(xml_path)
+        cmd = [scraper, 'devicetype', '-c', clusters_output_dir, '-nd', filename, xml_path]
+        if dry_run:
+            print(cmd)
+        else:
+            print(' '.join(cmd))
+            subprocess.run(cmd)
+
+    for filename in glob.glob(f'{device_type_dir}/*.adoc'):
```

Comment:
I guess this for loop needs to be adjusted a bit to not include the "sections_" files as well?